autotest/cpp/test_alg.cpp

```diff
@@ -273,4 +275,125 @@ TEST_F(test_alg, GDALAutoCreateWarpedVRT_alpha_band)
+}
+
+// Test GDALIsLineOfSightVisible() with single point dataset
+TEST_F(test_alg, GDALIsLineOfSightVisible_single_point_dataset)
+{
+    auto const sizeX = 1;
+    auto const sizeY = 1;
+    auto const numBands = 1;
+    GDALDatasetUniquePtr poDS(
+        GDALDriver::FromHandle(GDALGetDriverByName("MEM"))
+            ->Create("", sizeX, sizeY, numBands, GDT_Int8, nullptr));
+    ASSERT_TRUE(poDS != nullptr);
+
+    uint8_t val = 42;
+    auto pBand = poDS->GetRasterBand(1);
+    ASSERT_TRUE(pBand != nullptr);
+    ASSERT_TRUE(poDS->RasterIO(GF_Write, 0, 0, 1, 1, &val, 1, 1, GDT_Int8, 1,
+                               nullptr, 0, 0, 0, nullptr) == CE_None);
+    // Both points below terrain
+    EXPECT_FALSE(
+        GDALIsLineOfSightVisible(pBand, 0, 0, 0.0, 0, 0, 0.0, nullptr));
+    // One point below terrain
+    EXPECT_FALSE(
+        GDALIsLineOfSightVisible(pBand, 0, 0, 0.0, 0, 0, 43.0, nullptr));
+    // Both points above terrain
+    EXPECT_TRUE(
+        GDALIsLineOfSightVisible(pBand, 0, 0, 44.0, 0, 0, 43.0, nullptr));
+}
+
+// Test GDALIsLineOfSightVisible() with 10x10 default dataset
+TEST_F(test_alg, GDALIsLineOfSightVisible_default_square_dataset)
+{
+    auto const sizeX = 10;
+    auto const sizeY = 10;
+    auto const numBands = 1;
+    GDALDatasetUniquePtr poDS(
+        GDALDriver::FromHandle(GDALGetDriverByName("MEM"))
+            ->Create("", sizeX, sizeY, numBands, GDT_Int8, nullptr));
+    ASSERT_TRUE(poDS != nullptr);
+
+    auto pBand = poDS->GetRasterBand(1);
+    ASSERT_TRUE(pBand != nullptr);
+
+    const double x1 = 1;
+    const double y1 = 1;
+    const double x2 = 2;
+    const double y2 = 2;
+
+    // Both points are above terrain.
+    EXPECT_TRUE(
+        GDALIsLineOfSightVisible(pBand, x1, y1, 1.0, x2, y2, 1.0, nullptr));
+    // Flip the order, same result.
+    EXPECT_TRUE(
+        GDALIsLineOfSightVisible(pBand, x2, y2, 1.0, x1, y1, 1.0, nullptr));
+
+    // One point is below terrain.
+    EXPECT_FALSE(
+        GDALIsLineOfSightVisible(pBand, x1, y1, -1.0, x2, y2, 1.0, nullptr));
+    // Flip the order, same result.
+    EXPECT_FALSE(
+        GDALIsLineOfSightVisible(pBand, x2, y2, -1.0, x1, y1, 1.0, nullptr));
+
+    // Both points are below terrain.
+    EXPECT_FALSE(
+        GDALIsLineOfSightVisible(pBand, x1, y1, -1.0, x2, y2, -1.0, nullptr));
+    // Flip the order, same result.
+    EXPECT_FALSE(
+        GDALIsLineOfSightVisible(pBand, x2, y2, -1.0, x1, y1, -1.0, nullptr));
+}
+
+// Test GDALIsLineOfSightVisible() through a mountain (not a unit test)
+TEST_F(test_alg, GDALIsLineOfSightVisible_through_mountain)
+{
+    GDALAllRegister();
+    const std::string path = "/vsizip/vsicurl/https://terrain.ardupilot.org/"
+                             "SRTM1/ap_srtm1.zip/ap_srtm1.vrt";
+    const auto poDS = GDALDatasetUniquePtr(
+        GDALDataset::FromHandle(GDALOpen(path.c_str(), GA_ReadOnly)));
+    ASSERT_TRUE(poDS != nullptr);
```

Comment:
This will not work on builds without curl support, and when network is off. So (something like)

```suggestion
    if (!poDS )
    {
        GTEST_SKIP() << "Cannot open << path;
    }
```

Reply:
Alternatively we have a autotest/gdrivers/data/dted/n43_wgs72.dt0 dataset. Maybe you could try to find a good example with it, so we don't depend on curl? (for minimal builds the DTED driver might be unavailable, so we should gracefully skip if the dataset opening fails)

Reply:
Thanks! Yea, I just wanted to know a good dataset to use with sufficient enough height difference. I'll swap it over.